tests/zquantum/core/estimation/estimation_test.py

```diff
@@ -603,11 +629,62 @@ def test_evaluate_non_measured_estimation_tasks_fails_with_non_zero_shots(
             ),
         ],
         [
-            ExpectationValues(np.array([0])),
-            ExpectationValues(np.array([-2 * (np.cos(np.pi / 8) ** 2 - 0.5) * 2])),
+            ExpectationValues(np.array([0]), [np.array([[1]])], [np.array([[0.001]])]),
+            ExpectationValues(
+                np.array([-2 * (np.cos(np.pi / 8) ** 2 - 0.5) * 2]),
+                [np.array([[4]])],
+                [np.array([[0.002]])],
+            ),
         ],
     ),
 ]
+# When number_of_shots is high, the covariance will tend to zero. So we need extra
+# test cases to ensure covariance is being caluclated correctly.
+TEST_CASES_NONEIGENSTATES_WITH_LOW_NUMBER_OF_SHOTS = [
+    (
+        4
+        * [
+            EstimationTask(
+                IsingOperator("[Z0] + [Z1]"),
+                circuit=Circuit([X(0), X(1)]),
+                number_of_shots=10,
+            ),
+        ],
```

Comment:
There are 4 test examples. They seem to have the same "inputs", but different ref "outputs".

Is it because the actual input values don't matter? If so, WDYT about not hardcoding the disposable estimation tasks and passing `None` in the test instead?

Reply:
We need to pass some sort of `estimation_task` in order to avoid errors from `run_circuit_and_measure`. I updated the docstring in my new PR to make this more clear.